util/process_windows.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2019-2020 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+// +build windows
+
+package util
+
+import (
+	"os"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+// KillProcess kills a running OS process
+func KillProcess(pid int, _ os.Signal) error {
```

Comment:
This implementation is a great fit for the os.Signal SIGKILL. It doesn't emulate the SIGTERM correctly on windows.
Implementing it correctly require some extra work ( which can be deferred to a follow-up PR ):
1. Add a hidden console window to the algod and configure a SetConsoleCtrlHandler
2. In KillProcess, provide handling for SIGINT:
    Find the window that is associated with the process, and use GenerateConsoleCtrlEvent
3. In KillProcess, provide handling for SIGTERM:
    Find the window that is associated with the process, and send WM_QUIT using SendMessage.

Reply:
Go already sets up a SetConsoleCtrlHandler but the problem is how to send the signal to the target process.

Workarounds includes injecting a remote thread or attaching the target console (I don't like this approach)

Whatever the method, it will have some complexity.